src/Detection/Framework.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Utopia\Detector\Detection;
+
+use Utopia\Detector\Detection;
+
+abstract class Framework extends Detection
+{
+    protected string $packager = '';
+
+    public function __construct() {}
+
+    public function setPackager(string $packager): self
+    {
+        $this->packager = $packager;
+
+        return $this;
+    }
+
+    abstract public function getName(): string;
+
+    abstract public function getFiles(): array;
+
+    abstract public function getFileExtensions(): array;
```

Comment:
For all `array` types, let's use PHPdoc to document exact type, for example `@returns array<string>`

Reply:
Let's do this everywhere that we accept array as param, return array, or have a property on class that stores array